Revise login form

timogi opened the pull request:
The following changes were made:
- Login as guest button included in div and change style to default raised for better visibility
- Change of text on guest button to make it more clear for user that they do not need to provide credentials
- New componend to allow creation of divider with text
- Blurred background image to focus on login form
- Update prettierrc file to support latest node version

closes #282

Commit:
The following changes were made:
- Login as guest button included in div and change style to default raised for better visibility
- Change of text on guest button to make it more clear for user that they do not need to provide credentials
- New componend to allow creation of divider with text
- Blurred background image to focus on login form

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in eacaac8 according to the output
from Prettier.

Details: https://github.com/zskarte/zskarte-client/pull/308

deepsource-autofix[bot] committed:
refactor: autofix issues in 1 file

Resolved issues in src/app/state/i18n.service.ts with DeepSource Autofix

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in 60ea262 according to the output
from Prettier.

Details: https://github.com/zskarte/zskarte-client/pull/308

Commit:
fix empty class error

Commit:
autofix linting

Commit:
fix playwright test for button guest

timogi committed:
Merge pull request #313 from zskarte/dev

merge dev into feature branch

dfseifert committed:
chore: Update equality operator in protocolEntry

The commit replaces the equality operator with the strict equality operator in the protocolEntry helper. This ensures that compiled JavaScript code will check both type and value, improving comparison efficiency and code readability.

dfseifert committed:
chore: Simplify the 'sign' element assignment logic in protocolEntry

This commit refactors the 'sign' element assignment in the protocolEntry.ts file. The previous version was spread across several lines for readability, but it was deemed unnecessary. Thus, we simplified it into a single line without changing its logic, enhancing the overall code clarity.

dfseifert committed:
chore: Refactor coordinates format in projections helper

The coordinates format in the projections helper has been simplified and made more readable. The updated code now uses an if statement instead of a ternary operator to check the length of the coordinates, and uses template literals for improved clarity in the output string formation.

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in 99f718c according to the output
from Prettier.

Details: https://github.com/zskarte/zskarte-client/pull/308

dfseifert committed:
refactor: Refactor translate functions in projections helper

The translate functions in the projections helper file have been updated for better code clarity and readability. Variable names have been changed from x, y to more descriptive ones like latitude and longitude. A check for null or incorrect array length has also been added to prevent errors.

dfseifert committed:
Merge remote-tracking branch 'origin/feature/#282' into feature/#282

gobeli commented on src/assets/img/login-bg-blurred.png:
Evtl. bringt man die grösse von dem bild noch etwas runter, da es blurred ist sollte es auch nicht so eine Rolle spielen, wenn die Auflösung nicht top ist...

dfseifert committed:
chore: Refactor variable names in projections helper

Changed the variable names 'e' and 'n' to 'longitude' and 'latitude' in the function inside projections.ts. This change improves code readability and understanding, especially for developers not familiar with the original abbreviations.

Commit:
downsize login background image

gobeli commented on src/app/session/login/login.component.html:
```diff
@@ -39,12 +39,11 @@
-      <button class="login-button" mat-raised-button color="primary" (click)="login()">Login</button>
+      <stack [spacing]="5">
+        <button class="login-button" mat-raised-button color="primary" (click)="login()">Login</button>
+        <text-divider style="width: 100%">{{ i18n.get('or') }}</text-divider>
```
Vielleicht könnte man in der TextDivider Component gleicht auf `:host` die width auf 100% setzen. (https://angular.io/guide/component-styles#host)
